cmd/nginx/flags.go

```diff
@@ -303,16 +306,51 @@ https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-g
 		config.RootCAFile = *rootCAFile
 	}
 
+	var err error
 	if (nginx.MaxmindLicenseKey != "" || nginx.MaxmindMirror != "") && nginx.MaxmindEditionIDs != "" {
-		if err := nginx.ValidateGeoLite2DBEditions(); err != nil {
+		if err = nginx.ValidateGeoLite2DBEditions(); err != nil {
 			return false, nil, err
 		}
-		klog.InfoS("downloading maxmind GeoIP2 databases")
-		if err := nginx.DownloadGeoLite2DB(); err != nil {
+		klog.InfoS("trying to download maxmind GeoIP2 databases...")
+		defaultRetry := wait.Backoff{
+			Steps:    5,
+			Duration: nginx.MaxmindRetriesTimeout,
+			Factor:   1.5,
+			Jitter:   0.1,
+		}
+		if nginx.MaxmindRetriesTimeout == time.Duration(0) {
+			defaultRetry.Steps = 1
+		}
+
+		var lastErr error
+		retries := 0
+
+		err = wait.ExponentialBackoff(defaultRetry, func() (bool, error) {
+			err := nginx.DownloadGeoLite2DB()
```

Comment:
What if we move all of this to DownloadGeoLite2DB instead of adding to this main function in flags. 

https://github.com/kubernetes/ingress-nginx/blob/b1c8e3047ba31d7ea78f6e0915d187db75230ba5/internal/nginx/maxmind.go#L63

Also, I would like to keep here simple, if possible. I do not opose myself of changing DownloadGeoLite2DB signature to pass the delay and retries as arguments, but infact we should deal an error as an error and not rely too much on the syscall in this case :) 

If we have an error and the error happens here, we should just return to the caller function an error downloading the database, and parse this on the inner functions IMO

Reply:
Moved all the download retry logic into DownloadGeoLite2DB func